pkgs/development/tools/profiling/oprofile/default.nix

```diff
@@ -12,10 +12,14 @@ stdenv.mkDerivation rec {
     sha256 = "0nn4wfvwy4nii25y6lwlrnzx9ah4nz0r93yk7hswiy6wxjs10wc4";
   };
 
-  buildInputs = [ binutils zlib popt pkgconfig ]
+  buildInputs = [ binutils zlib popt pkgconfig linuxPackages.kernelHeaders ]
     ++ stdenv.lib.optionals withGUI [ qt4 ];
 
   configureFlags = [
+      "--build ${stdenv.system}"
```

Comment:
I'm sure in only one thing: oprofile wouldn't not build if the target and host platform options are not specified when '--with-kernel' switch is used. I can check if build is necessary.